mms/model_service/model_service.py

```diff
@@ -94,26 +94,33 @@ def inference(self, data):
             data to be sent back
         '''
         pre_start_time = time.time()
+
         data = self._preprocess(data)
-        infer_start_time = time.time()
 
         # Update preprocess latency metric
+        infer_start_time = time.time()
         pre_time_in_ms = (infer_start_time - pre_start_time) * 1000
         if self.model_name + '_LatencyPreprocess' in MetricsManager.metrics:
             MetricsManager.metrics[self.model_name + '_LatencyPreprocess'].update(pre_time_in_ms)
 
         data = self._inference(data)
-        data = self._postprocess(data)
 
+        post_start_ms = time.time()
         # Update inference latency metric
-        infer_time_in_ms = (time.time() - infer_start_time) * 1000
+        infer_time_in_ms = (post_start_ms - infer_start_time) * 1000
         if self.model_name + '_LatencyInference' in MetricsManager.metrics:
```

Comment:
Why do we need to do this over and over? Please keep adding to metrics object at the end.

Reply:
Sure. I can do the change. I am not sure why the if statements were there in the first place. I think these metrics are a must if metrics are enabled. I will dig a little deeper to see the reason why these are present.